be/src/storage/persistent_index.cpp

```diff
@@ -1943,15 +1964,31 @@ Status ShardByLengthMutableIndex::load(const MutableIndexMetaPB& meta) {
     std::string index_file_name = get_l0_index_file_name(_path, start_version);
     std::shared_ptr<FileSystem> fs;
     ASSIGN_OR_RETURN(fs, FileSystem::CreateSharedFromString(_path));
+    ASSIGN_OR_RETURN(auto read_file, fs->new_random_access_file(index_file_name));
     phmap::BinaryInputArchive ar(index_file_name.data());
     if (snapshot_size > 0) {
+        // check snapshot's crc32 checksum
+        const uint32_t expected_crc32c = snapshot_meta.crc32c();
+        // If expected crc32 is 0, which means no crc32 here, skip check.
+        // This may happen when upgrade from old version.
+        if (expected_crc32c > 0) {
+            uint32_t current_crc32c = 0;
+            RETURN_IF_ERROR(crc32c_of_file(read_file.get(), snapshot_off, snapshot_size, &current_crc32c));
+            if (current_crc32c != expected_crc32c) {
+                std::string error_msg = fmt::format(
```

Comment:
Add the `tablet_id` in the log

Reply:
tablet_id is already in `index_file_name`.